components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/validators/LogoutRequestValidator.java

```diff
@@ -0,0 +1,188 @@
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Validate the SAML version of the logout request.
+     *
+     * @param logoutRequest {@link LogoutRequest} object to be validated.
+     * @throws SAMLIdentityException If SAML version validation fails.
+     */
+    private void isSAMLVersionValid(LogoutRequest logoutRequest) throws SAMLIdentityException {
+
+        if (!(SAMLVersion.VERSION_20.equals(logoutRequest.getVersion()))) {
+            String notification = "Invalid SAML Version in Logout Request. SAML Version should be equal to 2.0";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLFedLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.VERSION_MISMATCH, notification);
+            throw new SAMLIdentityException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
+        }
+    }
+
+    /**
+     * Validate the issuer of the logout request.
+     *
+     * @param logoutRequest {@link LogoutRequest} object to be validated.
+     * @throws SAMLIdentityException If Issuer of the Logout Request validation fails.
+     */
+    private void isIssuerValid(LogoutRequest logoutRequest) throws SAMLIdentityException {
+
+        if (logoutRequest.getIssuer().getValue() != null) {
+            if (StringUtils.isBlank(logoutRequest.getIssuer().getFormat()) ||
+                    !(ISSUER_FORMAT.equals(logoutRequest.getIssuer().getFormat()))) {
+                String notification = "Invalid Issuer Format in the logout request";
+                if (log.isDebugEnabled()) {
+                    log.debug(notification);
+                }
+                samlMessageContext.setValidStatus(false);
+                String errorResponse = SAMLFedLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                        SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+                throw new SAMLIdentityException(notification, errorResponse, logoutRequest.getDestination(),
+                        samlMessageContext.getRelayState());
+            }
+        } else {
+            String notification = "Issuer value cannot be null in the Logout Request";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLFedLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+            throw new SAMLIdentityException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
```

Comment:
Rather than having this in the else block we can do the null check for logoutRequest.getIssuer().getValue() in the very beginning and increase the readability of this method.